internal/pkg/config/config.go

```diff
@@ -29,15 +29,23 @@ import (
 	"github.com/google/go-flow-levee/internal/pkg/config/regexp"
 )
 
-// FlagSet should be used by analyzers to reuse -config flag.
-var FlagSet flag.FlagSet
-var configFile string
+var (
+	// FlagSet should be used by analyzers to reuse -config flag.
+	FlagSet                    flag.FlagSet
+	configFile                 string
+	validFuncMatcherFields     []string
+	validSourceMatcherFields   []string
+	validFieldTagMatcherFields []string
+)
 
 func init() {
 	FlagSet.StringVar(&configFile, "config", "config.yaml", "path to analysis configuration file")
+	validFuncMatcherFields = []string{"package", "packagere", "receiver", "receiverre", "method", "methodre"}
```

Comment:
An honest, non-Socratic question: is there an advantage to having the map initialized in `init()` rather than at its `var` declaration?  The `FlagSet` didn't initialize in-line, so it had to live in an `init`, but arrays could be in-lined.  But more on the in the main comment.

Reply:
Updated